### ISSUE TITLE
revision (small) of simulated restarts

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#735]
When ECDF data are generated from data sets with unsuccessful and successful runs, simulated restarts are applied. They are applied "to all runs", also the successful runs. This seems to introduce unnecessary stochastics given we are (only) interested in the run-length distribution.  

The alternative: unless we have a specific interest in measuring dispersion, we should draw the first run, before to simulate the restart, with the (almost) exact proportion they appear in the data (up to `#bootstraps mod #runs`). Another advantage, besides the reduced stochastics: this should be overall a little bit faster.

**Comment by nikohansen** [comment]
Specifically, the code for an iterator over the run index could be
```python
class derandomized_index(object):
    def __init__(self, nb_of_indices, nb_of_iterations):
        self.indices = nb_of_indices
        self.maxiter = nb_of_iterations
        self.current_iteration = 0
        self.current_indices = []
    def __iter__(self):
        return self
    def next(self):
        if self.current_iteration >= self.maxiter:
            raise StopIteration
        self.current_iteration += 1
        if not len(self.current_indices):
            self.current_indices = np.random.permutation(range(self.indices))
        return self.current_indices.pop()
```

**Comment by nikohansen** [comment]
Status update: `DataSet.evals_with_simulated_restarts` (based `toolsstat.randint_derandomized`) is implemented with the idea to replace `toolsstat.drawSP` in `pprldmany`. Another proofread is desirable and the match of the interfaces needs to be carefully checked. A next step may be to wrap the `toolsstat.drawSP` interface around `DataSet.evals_with_simulated_restarts`, i.e. reimplement `toolsstat.drawSP` based on `DataSet.evals_with_simulated_restarts`, mainly for smooth transition and testing.

**Comment by nikohansen** [comment]
See also https://github.com/numbbo/coco/issues/1877

**Comment by nikohansen** [comment]
Implemented with commit 5e72e67a1.
